packages/number-field/src/NumberField.ts

```diff
@@ -319,7 +319,8 @@ export class NumberField extends TextfieldBase {
         if (isNaN(this.value)) {
             value = min;
         }
-        this._value = this.validateInput(value);
+        this.value = value;
```

Comment:
I'm not sure we want to trigger a render lifecycle in response to the `input` event. This _may_ be the cause of some of your failing tests.

Reply:
My bad, indeed. But, I still don't see this as expecting to update the `lastCommittedValue`, which is what happens when leveraging the setter liker this. _Maybe_ it would be worth it to do `this.inputElement.value = this.validateInput(value);` and then dispatching the `input` event on `this.inputElement` to tie into that `input` workflow? 🤔

Reply:
I think that works as well, although I see that the `requestUpdate` part of the value setter is not used when the component is by itself but only when it gets set from the outside (for example in sp-slider), if we go down this imperative path. This happens because `this._value` gets computed in `stepBy` and then `setValue` gets called in which the if guard will always return. I wonder if this is intended/best.
I think for this specific issue this imperative approach might be better but i truly wonder if long term the value setter and the `lastCommitedValue` could be slightly changed to have their meaning a little bit clearer.